[WD-17804] feat: Rebrand blog/index, blog/tag, blog/author and blog/topic pages

muhammad-ali-pk opened the pull request:
_**Note: Most of the code is copied over from canonical.com's blog, since majority of the design is similar, along with few differences**_

## Done

- Rebranded the following pages
  - blog/index
  - blog/topic
  - blog/tag
  - blog/author

## QA

- View the site in your web browser at: https://ubuntu-com-15214.demos.haus/blog
    - Be sure to test on mobile, tablet and desktop screen sizes
- Verify the design matches [figma](https://www.figma.com/design/CB22O7EPgyiUQheHVpOHQ9/ubuntu.com-blog) for the following pages
  - blog/index
  - blog/topic
  - blog/tag
  - blog/author

## Issue / Card

Fixes [WD-17804](https://warthogs.atlassian.net/browse/WD-17804)
         [WD-21614](https://warthogs.atlassian.net/browse/WD-21614)

## Help

[QA steps](https://discourse.canonical.com/t/qa-steps/152) - [Commit guidelines](https://discourse.canonical.com/t/commit-guidelines/148)


[WD-17804]: https://warthogs.atlassian.net/browse/WD-17804?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[WD-21614]: https://warthogs.atlassian.net/browse/WD-21614?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

webteam-app commented:
### [<img src='https://assets.ubuntu.com/v1/6baef514-ubuntu-circle-of-friends-large.svg' height=32 width=32> Demo</img>](https://ubuntu-com-15214.demos.haus)
### [<img src='https://assets.ubuntu.com/v1/e512b0e2-jenkins.svg' height=32 width=32> Jenkins </img>](https://jenkins.canonical.com/webteam/job/start-demo/6929)
### [<img src='https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg' height=32 width=32> demos.haus </img>](https://demos.haus)

mattea-turic commented:
Thank you @muhammad-ali-pk! They look great as per usual :) And since @eliman11 is away, I'll go over the UX review too:


---

## General:
### UX:
- [ ] If possible, could we order the chips so that the content type (i.e. article, webinar, case study, etc.) comes first, and then the rest can follow
- [ ] Could we display 14 articles across each index page. Right now, the topic and tag pages (e.g. [here](https://ubuntu-com-15214.demos.haus/blog/tag/smart-home)) are showing 12

---


## blog/index:
### Visual:
- [x] Perfect, no comments 

### UX:
- [x] Also no comments

---

## blog/topic:
### Visual:
**For all the pages:**
- [ ] Could you copy all the `<br>`s in the H2s as per the design pls


### UX:
**For the ["Web engineering and design team"](https://ubuntu-com-15214.demos.haus/blog/topics/design) page:**
- [ ] I noticed that this is the only topic without a period at the end of the H2 in the title – I updated it in the design; could you do the same in the webpage pls? I.e. "Discover the latest news and work from the web and design team."

**For "Desktop", "Cloud and Server", and "Snapcraft":**
- [ ] As per the style guide, we should be using the oxford comma – could you update these please (I made the change in Figma)

**For all the pages:**
- [ ] The links in the hero, so i.e. the "Learn more..." CTAs seem to be linking back to the same page they're on. Could you make sure that: IoT goes [here](https://ubuntu.com/internet-of-things); Desktop [here](https://ubuntu.com/desktop?_gl=1*o8wttx*_gcl_au*MTk4MTcwNDU2OS4xNzQ5ODI5MDQz); Cloud and Server [here](https://ubuntu.com/cloud); W&D [here](https://webteam.canonical.com/?_gl=1*pqzz98*_gcl_au*MTk4MTcwNDU2OS4xNzQ5ODI5MDQz*_ga*NjQzMjI3Mzk1LjE3NDk3MzQ0Mzk.*_ga_5LTL1CNEJM*czE3NTAzMjc2ODUkbzIwJGcxJHQxNzUwMzI3OTA5JGo0JGwwJGgw) – I'm not entirely sure about the rest, we may have to check with @eliman11 when she's back

**For "Juju", "MAAS", "Robotics", and "Snapcraft"**
- [ ] I'm not seeing these as in the Figma; i.e. Juju links [here](https://ubuntu-com-15214.demos.haus/blog/juju); snapcraft [here](https://ubuntu-com-15214.demos.haus/blog/snapcraft), and MAAS and Robotics give me 505 errors. If they just don't exist as topics, then it should be okay to ignore

---

## blog/tag
### Visual:
- [x] Perfect!


### UX:
- [ ] Can we have a number before "posts" in the hero to reflect the number of posts a given tag has. Right now I'm just seeing "posts" e.g. for [smart home](https://ubuntu-com-15214.demos.haus/blog/tag/smart-home). @eliman11 mentioned we're doing this for [author pages](https://canonical.com/blog/author/tkurek), so if this functionality can be replicated, that'd be amazing

---

## blog/author
### Visual:
- [x] Again, no comments here lol

### UX:
- [ ] Looks good :)

muhammad-ali-pk commented:
@mattea-turic Thank you for the detailed review. I have addressed your comments.

> I'm not seeing these as in the Figma; i.e. Juju links [here](https://ubuntu-com-15214.demos.haus/blog/juju); snapcraft [here](https://ubuntu-com-15214.demos.haus/blog/snapcraft), and MAAS and Robotics give me 505 errors. If they just don't exist as topics, then it should be okay to ignore

The topics in question exist at https://ubuntu-com-15214.demos.haus/blog/topics/juju, https://ubuntu-com-15214.demos.haus/blog/topics/snapcraft, https://ubuntu-com-15214.demos.haus/blog/topics/maas, and https://ubuntu-com-15214.demos.haus/blog/topics/robotics respectively.

> Can we have a number before "posts" in the hero to reflect the number of posts a given tag has. Right now I'm just seeing "posts" e.g. for [smart home](https://ubuntu-com-15214.demos.haus/blog/tag/smart-home). @eliman11 mentioned we're doing this for [author pages](https://canonical.com/blog/author/tkurek), so if this functionality can be replicated, that'd be amazing

> Could we display 14 articles across each index page. Right now, the topic and tag pages (e.g. [here](https://ubuntu-com-15214.demos.haus/blog/tag/smart-home)) are showing 12

Both of these needed modifications in the canonicalwebteam.blog package, so I have a [PR](https://github.com/canonical/canonicalwebteam.blog/pull/179) up to achieve both of these functionalities. Plus, You should still be able to verify this in the demo, since it's setup this way.
